### ISSUE TITLE
[3.x] Project manager does not parse `project.godot` files from 4.0 if the project uses `class_name`

### ISSUE THREAD

**Issue by jtnicholl** [godotengine/godot#66514]
### Godot version

v3.6.beta.custom_build [2157c50d1]

### System information

Linux (Fedora 36)

### Issue description

Normally, when a Godot 4 project is displayed in the Godot 3 project manager, the project appears with text grayed out, and attempting to open it fails with an error.
However, this doesn't happen if the Godot 4 project uses `class_name` anywhere in the project. Instead it shows as if the `project.godot` file was empty and will allow you to open it, only warning that the project doesn't specify its version (which is untrue).
The top project is a Godot 4 project using `class_name`, the bottom one is one that doesn't. The two in between are unrelated.
![Screenshot_20220927_173132](https://user-images.githubusercontent.com/44414418/192642017-3da178de-93b8-4235-95a5-92383fa178eb.png)
The following is printed for each project that fails to parse (the number `12` is occasionally a higher number):
```
ERROR: ConfigFile parse error at /path/to/project/project.godot:12: Unexpected character..
   at: _parse (core/io/config_file.cpp:280)
```

### Steps to reproduce

1. Open the project manager in 3.x
2. Add a Godot 4 project that uses `class_name`

### Minimal reproduction project

[ClassNames.zip](https://github.com/godotengine/godot/files/9659838/ClassNames.zip)

**Comment by gotnospirit** [comment]
Related to the parsing of the key `_global_script_classes` in `project.godot` file, because when removed there's no parse error

4.x:
```
config_version=5

_global_script_classes=[{
"base": "Node",
"class": &"TestClass",
"language": &"GDScript",
"path": "res://test_class.gd"
}]
```

3.x:
```
config_version=4

_global_script_classes=[ {
"base": "Node2D",
"class": "BaseLevel",
"language": "GDScript",
"path": "res://scenes/levels/base_level.gd"
}, {
```

**Comment by Calinou** [comment]
This is expected, as Godot 4.0 uses the `&"StringName"` notation which 3.x doesn't support. This is because StringName isn't exposed as a user-facing type in 3.x.

**Comment by jtnicholl** [comment]
Now that I think about it, should the class name declarations even be in `project.godot`? They're generated automatically from the script files, wouldn't it make more sense to store them in their own file somewhere like `.godot/global_script_classes`? Or is there a reason they need to be in version control?

> I'm not sure how this can be fixed, other than replacing `&"` with `"` in memory when loading the `project.godot` file in the 3.x project manager code.

Another option might be to have projects that fail to parse not display at all. Would also hide any project files that get corrupted somehow. I feel like treating the file as if it were empty is not a great way to handle this error.

**Comment by jtnicholl** [comment]
Is there any other situation in which the project.godot file can have StringNames in it? If not, this was fixed by #70557, but if so, the description needs to be updated.

**Comment by akien-mga** [comment]
Technically you could. I don't know if any of the existing project settings use StringNames, but you can add custom settings yourself which may.

But indeed the severity of this issue was reduced drastically, it's unlikely to be an issue for most users when 4.0 releases.
